python/ray/serve/autoscaling_metrics.py

```diff
@@ -0,0 +1,98 @@
+import threading
+import bisect
+from collections import defaultdict
+import time
+from typing import Callable, DefaultDict, Dict, List, Optional
+from dataclasses import dataclass, field
+
+
+def start_metrics_pusher(interval_s: float,
+                         collection_callback: Callable[[], Dict[str, float]],
+                         controller_handle):
+    """Start a background thread to push metrics to controller.
+
+    Args:
+        interval_s(float): the push interval.
+        collection_callback: a callable that returns the metric data points to
+          be sent over the the controller.
+        controller_handle: actor handle to Serve controller.
```

Comment:
for unit testing it's probably better to make an interface wrapping this instead of passing in an actor handle directly